In [1]:
import pandas as pd

df = pd.read_csv('result.csv')
df.head()

id  ...                                               text
0  1389953924879110144  ...  abis minum kopi perasaan mesti gelisah gatau k...
1  1389952303407329283  ...          sumpah ya dari pagi rasanya gelisah terus
2  1389952181676122116  ...  Catia hari sabtu.Presntasi Statin hari sabtu.....
3  1389951640610902020  ...                                     tbtb gelisah 🤕
4  1389950790266023936  ...  Memasuki jam jam rawan... Seliweran mulu di ga...

[5 rows x 7 columns]

In [2]:
df = df.filter(['id', 'text'])
df = df[df['text'].map(lambda x: len(x.split()) >= 3)]
df.head()

id                                               text
0  1389953924879110144  abis minum kopi perasaan mesti gelisah gatau k...
1  1389952303407329283          sumpah ya dari pagi rasanya gelisah terus
2  1389952181676122116  Catia hari sabtu.Presntasi Statin hari sabtu.....
3  1389951640610902020                                     tbtb gelisah 🤕
4  1389950790266023936  Memasuki jam jam rawan... Seliweran mulu di ga...

In [3]:
len(df)

2751

In [4]:
df = df.head(1000)
len(df)

1000

In [5]:
df

id                                               text
0     1389953924879110144  abis minum kopi perasaan mesti gelisah gatau k...
1     1389952303407329283          sumpah ya dari pagi rasanya gelisah terus
2     1389952181676122116  Catia hari sabtu.Presntasi Statin hari sabtu.....
3     1389951640610902020                                     tbtb gelisah 🤕
4     1389950790266023936  Memasuki jam jam rawan... Seliweran mulu di ga...
...                   ...                                                ...
1019  1389076387684700162  Moots kalau berusaha tidur hadap kanan, hadap ...
1020  1389075326664286209  aku diem aja mbak nya cemas gmn kl aku gerak d...
1021  1389074928893267971  Tenang saja, semua akan baik baik saja. Tidak ...
1022  1389074441762611201  Tak sukanya orang beli baju berebut ni lepastu...
1023  1389074401883086850  Kalau pagi2 resah, gelisah, anxious itu tanda ...

[1000 rows x 2 columns]

Three annotators, so each of them will get 300 tweets + 100 shared tweets. In total:

$$3\times 300 + 100 = 1000$$

In [25]:
split1, split2, split3, share = df[0:300], df[300:600], df[600:900], df[900:]

In [ ]:
split1.to_csv('split1.csv', index=False)
split2.to_csv('split2.csv', index=False)
split3.to_csv('split3.csv', index=False)
share.to_csv('share.csv', index=False)

# Creating DATD dataset

Merging back responses

In [26]:
def load_response(response_name):
  res = []
  with open('{}.response'.format(response_name)) as f:
    res = [int(r.strip()) for r in f.readlines()]
  return res

In [27]:
rsplit1, rsplit2, rsplit3 = load_response('split1'), load_response('split2'), load_response('split3')

In [28]:
rshare1, rshare2, rshare3 = load_response('share-1'), load_response('share-2'), load_response('share-3')

In [30]:
print(rshare1)
print(rshare2)
print(rshare3)

[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0]
[1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0]
[1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0]


In [32]:
rshare = []

# three annotators, so tie is not possible
for i in range(len(rshare1)):
  zero = 0
  one = 0
  if rshare1[i] == 0:
    zero += 1
  else:
    one += 1
  
  if rshare2[i] == 0:
    zero += 1
  else:
    one += 1
  
  if rshare3[i] == 0:
    zero += 1
  else:
    one += 1
  
  if zero > one:
    rshare.append(0)
  else:
    rshare.append(1)

print(rshare)

[1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0]


In [35]:
annotated_share = share.assign(label=rshare)

In [36]:
annotated_split1 = split1.assign(label=rsplit1)
annotated_split2 = split2.assign(label=rsplit2)
annotated_split3 = split3.assign(label=rsplit3)

In [41]:
annotated_dataset = pd.concat([annotated_split1, annotated_split2, annotated_split3, annotated_share], axis=0)

In [42]:
annotated_dataset

id  ... label
0     1389953924879110144  ...     0
1     1389952303407329283  ...     1
2     1389952181676122116  ...     1
3     1389951640610902020  ...     0
4     1389950790266023936  ...     0
...                   ...  ...   ...
1019  1389076387684700162  ...     0
1020  1389075326664286209  ...     0
1021  1389074928893267971  ...     0
1022  1389074441762611201  ...     0
1023  1389074401883086850  ...     0

[1000 rows x 3 columns]

In [45]:
from collections import Counter

label_counter = Counter(annotated_dataset['label'])
print(label_counter)

Counter({0: 591, 1: 409})


Dropping `id` because it's not relevant for classification

In [ ]:
annotated_dataset = annotated_dataset.drop('id', axis=1)

In [49]:
print(annotated_dataset)

                                                   text  label
0     abis minum kopi perasaan mesti gelisah gatau k...      0
1             sumpah ya dari pagi rasanya gelisah terus      1
2     Catia hari sabtu.Presntasi Statin hari sabtu.....      1
3                                        tbtb gelisah 🤕      0
4     Memasuki jam jam rawan... Seliweran mulu di ga...      0
...                                                 ...    ...
1019  Moots kalau berusaha tidur hadap kanan, hadap ...      0
1020  aku diem aja mbak nya cemas gmn kl aku gerak d...      0
1021  Tenang saja, semua akan baik baik saja. Tidak ...      0
1022  Tak sukanya orang beli baju berebut ni lepastu...      0
1023  Kalau pagi2 resah, gelisah, anxious itu tanda ...      0

[1000 rows x 2 columns]


In [61]:
annotated_dataset.to_csv('datd.csv', index=False)

# Creating DATD+Rand dataset

In [50]:
annotated_dataset[annotated_dataset['label'] == 1]

text  label
1             sumpah ya dari pagi rasanya gelisah terus      1
2     Catia hari sabtu.Presntasi Statin hari sabtu.....      1
11    Pernah gasi ngerasain tiap hari gelisah karena...      1
14    Aku makin gelisah, dan aku menangis sejadi jad...      1
15        ko aku dekdekan kenapa ya??aku gelisah banget      1
...                                                 ...    ...
1007  2020-2021 adalah tahun sulit dimana aku kehila...      1
1008                     gelisah, padahal gitu doang bd      1
1014                              Kok gw jadi cemas yak      1
1015  Entah kenapa tiba-tiba rasanya gundah, resah, ...      1
1018                Het gelisah bet ini ngapa tau wkwkk      1

[409 rows x 2 columns]

Based on the paper, DATD+Rand contained $10.5\%$ positive instances, so we will solve

$$\dfrac{409}{409+x} = 0.105$$

and we get $x = \lfloor 3486.24 \rfloor \approx 3486$

In [59]:
random_df = pd.read_csv('random_dataset.csv')
random_df = random_df.filter(['text'])
random_df['label'] = 0
random_df.head(3486)

text  label
0     Gr gr mimpiin mantan crush jdi gamon gue ya Allah      0
1                                  wkwkwk apaansi lawak      0
2     Cuti2 pun standby depan pc 😅 sebab office tak ...      0
3            Alhamdulillah meskipun hanya numpang lewat      0
4     konsep teaser pernikahan pake lagu day6 - only...      0
...                                                 ...    ...
3481  ntar dlu ya ma lipat bajunya ak lagi naikin ht...      0
3482                            Ah mangkok nye aku 😭😭😭😭      0
3483  “Walaupun aQ hrs membahayakan hidupku, aQ hrs ...      0
3484                             teel isinya perbucinan      0
3485         Lamak na status proses pengkreditan tok 😭😭      0

[3486 rows x 2 columns]

In [60]:
annotated_random = pd.concat([annotated_dataset[annotated_dataset['label'] == 1], random_df.head(3486)], axis=0)
annotated_random

text  label
1             sumpah ya dari pagi rasanya gelisah terus      1
2     Catia hari sabtu.Presntasi Statin hari sabtu.....      1
11    Pernah gasi ngerasain tiap hari gelisah karena...      1
14    Aku makin gelisah, dan aku menangis sejadi jad...      1
15        ko aku dekdekan kenapa ya??aku gelisah banget      1
...                                                 ...    ...
3481  ntar dlu ya ma lipat bajunya ak lagi naikin ht...      0
3482                            Ah mangkok nye aku 😭😭😭😭      0
3483  “Walaupun aQ hrs membahayakan hidupku, aQ hrs ...      0
3484                             teel isinya perbucinan      0
3485         Lamak na status proses pengkreditan tok 😭😭      0

[3895 rows x 2 columns]

In [62]:
annotated_random.to_csv('datd_rand.csv', index=False)